src/Vue/Resources/doc/index.rst

```diff
@@ -113,7 +114,11 @@ The event ``vue:before-mount`` is called before a component is mounted on the pa
         app.use(router);
     });
 
-The event ``vue:before-mount`` is called when a component has been mounted on the page:
+.. note::
+   When using Vue Router, it is recommended to use "hash" (or "memory") history mode,
+   to prevent your Vue routes from being served through Symfony controllers.
```

Comment:
true history mode WOULD work, however, if you had some sort of a "catch all" route that executed the same controller, template and Vue component, right? For example, you are building a Vue app just for a "survey" section, for example. So you have a route:

```php
#Route('/survey/{path<.+>}')
public function survey($path = ''): Response
{
    // render the template
}
```

Assuming all Vue links point to URLs that start with `/survey`, this would a proper way to allow for the usage of the real history mode I think?

I like your note - and it made me think "Can we help instruct users also on how to use real history?"

Reply:
@weaverryan

> Can we help instruct users also on how to use real history?

Good idea, thanks. I added an extra section on using web history mode with a catch all route